src/components/card/card.tsx

```diff
@@ -87,16 +99,30 @@ export class Card {
     @Event()
     public actionSelected: EventEmitter<ActionBarItem>;
 
+    @Element() private element: HTMLElement;
+    private the3dElementBounds: DOMRect;
+
+    public componentDidLoad() {
+        const the3dElement = this.element.shadowRoot.querySelector('section');
+        the3dElement.addEventListener('mouseenter', this.handleMouseEnter);
+        the3dElement.addEventListener('mouseleave', this.handleMouseLeave);
+    }
```

Comment:
Hmm… I wonder if these event listeners need to be removed manually, or if they are removed automatically when the component is destroyed because they are applied to an internal element… 🤔

We need to not leak memory. The web client leaks _way_ too much memory already. I don't feel I can take the time to look that up right now though.

Reply:
> Hmm… I wonder if these event listeners need to be removed manually, or if they are removed automatically when the component is destroyed because they are applied to an internal element… 🤔

The overlord said:
> To ensure that there are no memory leaks, it's a good practice to manually remove event listeners when the component is destroyed. You can achieve this by adding a `disconnectedCallback` method to your component and removing the event listeners there.

So I added a fixup

Reply:
I don't understand why we need to register these in `componentDidLoad` when we already attach them using `onMouseEnter` and `onMouseLeave` in the render function?